### PR TITLE
Deploy Vault servers in an autoscale group with KMS auto-unseal

### PR DESCRIPTION
**Pull request by blarghmatey**
In order to support a scalable method of managing secrets across our infrastructure we need an easy way to deploy Vault clusters. This adds code to build Vault server images that can be deployed in an autoscale group without requiring any additional systems being in place beyond AWS services.

### CODE REVIEW

**Comment by blarghmatey** [comment]
Closes #149

File: src/bilder/components/hashicorp/vault/models.py [hunk 1]
```diff
@@ -67,12 +70,17 @@ class VaultAgentCache(FlexibleBaseModel):
     use_auto_auth_token: Union[str, bool] = True
 
 
-class VaultSealConfig(FlexibleBaseModel, abc.ABC):
-    pass  # noqa: WPS420, WPS604
+class VaultAwsKmsSealConfig(FlexibleBaseModel):
```
Comment (shaidar) [inline src/bilder/components/hashicorp/vault/models.py:73]: What does the `FlexibleBaseModel` provide us? I see the class definition, but not sure I understand what it's helping us with

Comment (blarghmatey) [reply src/bilder/components/hashicorp/vault/models.py:73]: It lets us add additional, arbitrary attributes to the class object without it throwing an exception. The intent being that we can add configuration parameters for classes that aren't completely fleshed out to match what the project is expecting for configuration inputs.

**Comment by blarghmatey** [comment]
Going to merge this as-is and do the review post-hoc to find and follow-up with any style/organizational changes.

**Pull request merged**